notification/src/handler/notification/notification.handler.js

```diff
@@ -181,7 +181,7 @@ function getAddTransactionUpdateAction (type, state, amount, currency) {
 
 async function getPaymentByMerchantReference (merchantReference, ctpClient) {
   try {
-    const query = ctpClient.builder.payments.where(`custom(fields(merchantReference="${merchantReference}"))`)
+    const query = ctpClient.builder.payments.where(`key="${merchantReference}"`)
```

Comment:
You could also get payment by key directly without `where` statement: https://docs.commercetools.com/http-api-projects-payments#get-payment-by-key

Reply:
@butenkor @lojzatran 
I've one question regarding commit https://github.com/commercetools/commercetools-adyen-integration/pull/395/commits/ac46b59fb32ed132caffc587b2a8404011f41234 handling not found exceptions on `fetchByKey`, 

https://github.com/commercetools/commercetools-adyen-integration/blob/ac46b59fb32ed132caffc587b2a8404011f41234/notification/src/handler/notification/notification.handler.js#L187-L189

also in code we are using `fetchById` which 404 is not handled for 404, do you have any suggestions? 

https://github.com/commercetools/commercetools-adyen-integration/blob/ac46b59fb32ed132caffc587b2a8404011f41234/notification/src/handler/notification/notification.handler.js#L67-L68

Reply:
I don't see a problem. First one is fetching by key when you receive a notification from Adyen. In this case, it could be 404 since we get the notification from the external system.

The other one is when you're retrying on update error. The other way I could see that you get 404 is if sth deletes the payment or some platform error, which should not happen.

So I think it's ok like we have it now.

Reply:
In case of 404, we could log out an info or warning. Not sure if it is very useful though.

Reply:
we already have: 
https://github.com/commercetools/commercetools-adyen-integration/blob/ac46b59fb32ed132caffc587b2a8404011f41234/notification/src/handler/notification/notification.handler.js#L30-L35